lib/plugin.js

```diff
@@ -27,19 +27,33 @@ function plugin (pluginOptions) {
     ...pluginOptions
   }
 
+  const optionsTypeValidations = {
+    sharedSecret: 'string',
+    verificationError: 'string',
+    verificationErrorMessage: 'string',
+    extractSignature: () => {},
+    constructSignatureString: () => {},
+    getDigest: () => {},
+    digestEncoding: 'string',
+    getAlgorithm: () => {},
+    algorithmMap: {},
+    getSignatureEncoding: () => {}
+  }
+
+  for (const [key, value] of Object.entries(optionsTypeValidations)) {
+    if (!options[key] || typeof options[key] !== typeof value) {
+      throw new Error(`${key} option does not exist or is not of type ${typeof value}`)
+    }
+  }
+
```

Comment:
@jkirkpatrick24 Is this what you were thinking for plugin options validation?

Reply:
It will be something like this for sure.I would like to keep this PR scoped only to the algorithm stuff for now though. Lets address the validations in another PR 👍

Reply:
Ok. I've removed these changes from this branch.